sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -54,10 +54,31 @@ private case class MySQLDialect() extends JdbcDialect with SQLConfHelper with No
     override def visitExtract(field: String, source: String): String = {
       field match {
         case "DAY_OF_YEAR" => s"DAYOFYEAR($source)"
-        case "YEAR_OF_WEEK" => s"EXTRACT(YEAR FROM $source)"
+        // The second argument of WEEK is mode, which controls the first day of the week, the range
+        // of return value, config of week 1 (with a Sunday in this year, with 4 or more days this
+        // year, with a Monday in this year).
+        // The mode 3 means the first day of the week is Monday, the range of return value is 1-53
+        // and week 1 is the first week with 4 or more days this year. In the other word, mode 3
+        // has the same semantics as ISO week.
+        case "WEEK" => s"WEEK($source, 3)"
```

Comment:
is it supported in all mysql versions?

Reply:
Yes. It is supported from MySQL 5.7 to MySQL 9.2(latest)